internal/pkg/agent/migration/migrate_config.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package migration
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/storage"
+)
+
+// MigrateToEncryptedConfig will copy content from the store specified in unencryptedConfigPath
+// to the encrypted store specified in encryptedConfigPath if it doesn't exist already.
+// This function is intended to be called during startup when *.yml files exist from a previous version
+// but no corresponding *.enc file has been generated yet.
+// Notes:
+//   - The contents from the unencrypted file will be copied as a byte stream without any transformation.
+//   - The function will not perform any operation if the encryptedConfigPath already exists and it's not empty to avoid overwrites.
+//   - If neither the encrypted file nor the unencrypted file exist this call is a no-op
+func MigrateToEncryptedConfig(l *logp.Logger, unencryptedConfigPath string, encryptedConfigPath string) error {
+	encStat, encFileErr := os.Stat(encryptedConfigPath)
+
+	if encFileErr != nil && !errors.Is(encFileErr, fs.ErrNotExist) {
+		return errors.New("error checking for existence of %s: %v", encryptedConfigPath, encFileErr)
```

Comment:
@cmacknz What do you think about this behavior? 
This came up while fixing the unit tests of this PR for windows: if we have an access error we got a SIGSEGV  on the check below
`isEncryptedConfigEmpty := errors.Is(encFileErr, fs.ErrNotExist) || encStat.Size() == 0`

Returning an error immediately will stop the agent startup , not returning an error here can probably bring to other failures further down the start sequence and I am not sure what is preferable (since both of those are bad in their own way). 
What's your take on this ?

Reply:
Returning an error is definitely better than panicing with a segfault.

What causes this situation, trying to migrate an encrypted file that doesn't exist? In that case there is no migration to do, is that truly an error? Is it valid for the files here to not exist yet?